pkg/api/types.go

```diff
@@ -1309,3 +1317,27 @@ type ResourceQuotaList struct {
 	// Items is a list of ResourceQuota objects
 	Items []ResourceQuota `json:"items"`
 }
+
+// Secret holds secret data of a certain type
+type Secret struct {
+	TypeMeta   `json:",inline"`
+	ObjectMeta `json:"metadata,omitempty"`
+
+	Data map[string][]byte `json:"data,omitempty"`
```

Comment:
Ahh, I see.  The comment led me astray.  I though people you were taking strings and base64ing them and then storing that in []byte.  But you're storing arbitrary secret data in []byte, and JSON serializes it to base64.  I'm going to update the comments in my PR, see if it is any better when I am done. :)